### ISSUE TITLE
Loading Pre-Annotated Regions for Visium Analysis

### ISSUE THREAD

**Issue by abadgerw** [pachterlab/voyager#41]
I have pre-annotated regions in QuPath and exported them as geoJSON files. Can these be loaded directly or converted in some way so that I may perform spatially-aware DE of my Visium data?

**Comment by lambdamoses** [comment]
Yes, I also do that. You can use `sf::st_read()` to read GeoJSON file as a `sf` data frame, make sure to set `crs = NA`; CRS is the different ways to project the globe onto a 2D map, which is irrelevant to histological space. The catch though is if the QuPath annotations have the same unit as the Visium coordinates. If you did the annotations with the full resolution image and the Visium coordinates are also in full res pixels, then they should match. If you have terra >= 1.17.83, then the image will appear flipped compared to QuPath so y would increase going up just as in the geometries so the geometries should match the images as well. Here's some pseudocode, say you saved your annotations in `annotations.geojson`:

```
library(sf)
g <- st_read("annotations.geojson", crs = NA)
annotGeometry(sfe, "annotations") <- g # suppose you have one sample
```

If you have multiple samples in the SFE object, you should add a column named "sample_id" specifying which geometry belongs to which sample.

**Comment by abadgerw** [comment]
Thanks, @lambdamoses this is very helpful! I will give it a try. Once I have created an SFE object that has the geojson read annotations in annotGeometries, can the object be converted to a Seurat object without losing this information or can the geojson read by st_read be directly loaded into a Seurat object or read by the CreateSegmentation function?

**Comment by abadgerw** [comment]
@lambdamoses I have been trying this but run into the issue that the full resolution .svs image and the visium image are of different pixel sizes so the GeoJSON doesn't line up with the visium image. Is there an easy way to rescale this?

**Comment by lambdamoses** [comment]
Does the svs file have metadata on how many microns per pixels? If you convert the units to microns it may line up.
